pagerduty/resource_pagerduty_event_orchestration_path_router_test.go

```diff
@@ -0,0 +1,403 @@
+					resource.TestCheckResourceAttr(
+						"pagerduty_event_orchestration_router.router", "sets.0.rules.#", "0"),
+					testAccCheckPagerDutyEventOrchestrationPathRouteToMatch(
+						"pagerduty_event_orchestration_router.router", "pagerduty_service.bar", true),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckPagerDutyEventOrchestrationPathDestroy(s *terraform.State) error {
+	client, _ := testAccProvider.Meta().(*Config).Client()
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "pagerduty_event_orchestration_path_router" {
+			continue
+		}
+
+		orch, _ := s.RootModule().Resources["pagerduty_event_orchestration.orch"]
+
+		if _, _, err := client.EventOrchestrationPaths.Get(orch.Primary.ID, "router"); err == nil {
+			return fmt.Errorf("Event Orchestration Path still exists")
+		}
+	}
+	return nil
+}
+
+func testAccCheckPagerDutyEventOrchestrationPathExists(rn string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[rn]
+		if !ok {
+			return fmt.Errorf("Not found: %s", rn)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Event Orchestration Path Type is set")
+		}
+
+		orch, _ := s.RootModule().Resources["pagerduty_event_orchestration.orch"]
+		client, _ := testAccProvider.Meta().(*Config).Client()
+		found, _, err := client.EventOrchestrationPaths.Get(orch.Primary.ID, "router")
+
+		if err != nil {
+			return fmt.Errorf("Orchestration Path type not found: %v for orchestration %v", "router", orch.Primary.ID)
+		}
+		if found.Type != "router" {
+			return fmt.Errorf("Event Orchrestration path not found: %v - %v", "router", found)
+		}
+		return nil
+	}
+}
+
+func createBaseConfig(t, ep, s, o string) string {
+	return fmt.Sprintf(`
+	resource "pagerduty_team" "foo" {
+		name = "%s"
+	}
+
+	resource "pagerduty_user" "foo" {
+		name        = "user"
+		email       = "user@pagerduty.com"
+		color       = "green"
+		role        = "user"
+		job_title   = "foo"
+		description = "foo"
+	}
+
+	resource "pagerduty_escalation_policy" "foo" {
+		name        = "%s"
+		description = "bar"
+		num_loops   = 2
+
+		rule {
+			escalation_delay_in_minutes = 10
+			target {
+				type = "user_reference"
+				id   = pagerduty_user.foo.id
+			}
+		}
+	}
+
+	resource "pagerduty_service" "bar" {
+		name = "%s"
+		escalation_policy       = pagerduty_escalation_policy.foo.id
+
+		incident_urgency_rule {
+			type = "constant"
+			urgency = "high"
+		}
+	}
+
+	resource "pagerduty_event_orchestration" "orch" {
+		name = "%s"
+		team {
+			id = pagerduty_team.foo.id
+		}
+	}
+	`, t, ep, s, o)
+}
+
+func testAccCheckPagerDutyEventOrchestrationPathConfigNoRules(t, ep, s, o string) string {
+	return fmt.Sprintf("%s%s", createBaseConfig(t, ep, s, o),
+		`resource "pagerduty_event_orchestration_router" "router" {
+		type = "router"
+		parent {
+			id = pagerduty_event_orchestration.orch.id
+		}
+		catch_all {
+			actions {
+				route_to = "unrouted"
+			}
+		}
+		sets {
+			id = "start"
+		}
```

Comment:
Sets with no rules property is meant to have the effect of creating a set with no rules. The Update handler, API Client knows to pass an empty list for rules property and the API returns with rules as an empty list.